### ISSUE TITLE
Animation editor key snapping is rounded to some predetermined value instead of using exact snap value

### ISSUE THREAD

**Issue by UnfavorableEnhancer** [godotengine/godot#98609]
### Tested versions

- Reproducible in : 4.4.dev3, 4.4.dev2
- Not reproducible in : 4.2.2.stable

### System information

Windows 11 - Godot v4.4.dev3

### Issue description

When you set time snap value in animation editor and you try to create or move key, the result key time value is rounded to some predetermined value instead of using exact number to snap.
Some examples:
- If snap value is 0.15, key is actually snapped to value 0.1429
- If snap value is 0.225 or 0.275, key is actually snapped to value 0.25
- If snap value is 0.3 or 0.4, key is actually snapped to value 0.3333333...
- If snap value is from 0.45 to 0.75, key is actually snapped to value 0.5
- If snap value is bigger than 0.75, key is actually snapped to value 1
The same rules apply for any integer value like 2.225 is snapped to 2.25, or 4.3 is snapped to 4.33333...

Note that timeline cursor doesn't have this issue and it snaps correctly to exact snap value, no matter how much decimal part is.

### Steps to reproduce

1.) Create AnimationPlayer, some Sprite or ColorRect, and create some simple moving animation for it
2.) Try to set snapping value to 0.25, 0.3, 0.425, 0.45 and 0.75 and try to create new key or move existing one and see that it snaps to incorrect value, while timeline cursor is moving correctly (when timeline cursor snap is enabled)

### Minimal reproduction project (MRP)

[MRP.zip](https://github.com/user-attachments/files/17541034/MRP.zip)

**Comment by AThousandShips** [comment]
Sounds like:
* https://github.com/godotengine/godot/issues/97832

Which has been fixed, if you are able to [build from source](https://docs.godotengine.org/en/latest/contributing/development/compiling/) you can test the latest version, or wait for dev4 which should be out soon

**Comment by UnfavorableEnhancer** [comment]
> Sounds like:
> 
> * [Animation track editor no longer snapping to intervals of 0.0001 when using seconds snapping mode. #97832](https://github.com/godotengine/godot/issues/97832)
> 
> Which has been fixed, if you are able to [build from source](https://docs.godotengine.org/en/latest/contributing/development/compiling/) you can test the latest version, or wait for dev4 which should be out soon

Just built sources from master branch and no it's still broken

**Comment by UnfavorableEnhancer** [comment]
![2024-10-30 13_15_13-animation_track_editor cpp - godot - Visual Studio Code](https://github.com/user-attachments/assets/5e0b51a2-dd39-4509-beef-a068ece4e2f7)
ok I've managed to fix this bug by commenting this section of code (which doesn't make any sence tbh) and just setting "snap_unit" straight to the "step" value

**Comment by TokageItLab** [comment]
I assume half of this problem is a specification rather than a bug.

When importing an external 3D animation, the default snap value is `1/30`, which is an unreasonable number, `0.0333`.

The problem here is that simple truncation and snapping will not allow snapping to 1.0s, so we use `modf()` snapping with https://github.com/godotengine/godot/pull/98826 and https://github.com/godotengine/godot/pull/96328 to snap every 1.0 interval.

There are two possible approaches:

1. Set the default for all animation modes to FPS Mode to prevent snapping to 1.0 when 0.333 is accumulated
    - In other words, needs to add a change to #98826 to all animation default to FPS Mode
2. Ignore the accumulation error issue when snapping above 1.0 and document that
    - In other words, #98826 only applies when step is 1.0 or higher and needs to add documentation

I believe that the default change by approach 1 should not be done, at least not during Godot5, since it could break projects that use the existing 0.333 snap. So we should probably choose approach 2.
